### PR TITLE
Peek into world-readable rooms from spotlight

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/vector-im/element-web/issues/22862

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Peek into world-readable rooms from spotlight ([\#9115](https://github.com/matrix-org/matrix-react-sdk/pull/9115)). Fixes vector-im/element-web#22862.

### CODE REVIEW

**Comment by t3chguy** [comment]
This won't fix that bug, the buttons still say `Join` when they're just showing you a splash preview with a Join button and no peek. I don't think you need to specify `peek: true` - it is the default

**Comment by SimonBrandner** [comment]
> This won't fix that bug, the buttons still say `Join` when they're just showing you a splash preview with a Join button and no peek. I don't think you need to specify `peek: true` - it is the default

I actually thought that this would specify `peek: false` when we cannot peek - we'll try to join right away?

**Comment by t3chguy** [comment]
I think you need to specify `auto_join: true` for that

**Comment by SimonBrandner** [comment]
> I think you need to specify `auto_join: true` for that

Ah, ok, I somehow thought that if we weren't peeking into the room, we would join it... Sorry

File: src/components/views/dialogs/spotlight/SpotlightDialog.tsx [hunk 1]
```diff
@@ -510,6 +510,7 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
             metricsViaKeyboard: viaKeyboard,
             room_id: room.roomId,
             room_alias: room.roomAlias,
+            auto_join: room.autoJoin,
```
Comment (t3chguy) [inline src/components/views/dialogs/spotlight/SpotlightDialog.tsx:513]: might be worth to also pass `peek: false` in this case to save it possibly trying to peek

**Review by t3chguy — APPROVED** [review]
Great, works as expected!

**Pull request merged**